aggregator/src/datastore.rs

```diff
@@ -1608,6 +1673,70 @@ impl<C: Clock> Transaction<'_, C> {
             .collect()
     }
 
+    /// get_report_aggregations_for_task retrieves all report aggregations associated with a given
+    /// task.
+    #[cfg(feature = "test-util")]
+    pub async fn get_report_aggregations_for_task<const L: usize, A: vdaf::Aggregator<L>>(
+        &self,
+        vdaf: &A,
+        role: &Role,
+        task_id: &TaskId,
+    ) -> Result<Vec<ReportAggregation<L, A>>, Error>
+    where
+        for<'a> A::PrepareState: ParameterizedDecode<(&'a A, usize)>,
+        A::OutputShare: for<'a> TryFrom<&'a [u8]>,
+        for<'a> &'a A::AggregateShare: Into<Vec<u8>>,
+    {
+        let stmt = self
+            .tx
+            .prepare_cached(
+                "SELECT
+                    aggregation_jobs.aggregation_job_id, client_reports.report_id,
+                    client_reports.client_timestamp, report_aggregations.ord,
+                    report_aggregations.state, report_aggregations.prep_state,
+                    report_aggregations.prep_msg, report_aggregations.out_share,
+                    report_aggregations.error_code
+                FROM report_aggregations
+                JOIN client_reports ON client_reports.id = report_aggregations.client_report_id
+                JOIN aggregation_jobs ON aggregation_jobs.id = report_aggregations.aggregation_job_id
+                WHERE aggregation_jobs.task_id = (SELECT id FROM tasks WHERE task_id = $1)",
+            )
+            .await?;
+        self.tx
+            .query(&stmt, &[/* task_id */ &task_id.as_ref()])
+            .await?
+            .into_iter()
+            .map(|row| {
+                let aggregation_job_id_bytes: [u8; AggregationJobId::LEN] = row
+                    .get::<_, Vec<u8>>("aggregation_job_id")
+                    .try_into()
+                    .map_err(|err| {
+                        Error::DbState(format!(
+                            "couldn't convert aggregation_job_id value: {err:?}"
+                        ))
+                    })?;
+                let aggregation_job_id = AggregationJobId::from(aggregation_job_id_bytes);
+
+                let report_id_bytes: [u8; ReportId::LEN] = row
+                    .get::<_, Vec<u8>>("report_id")
+                    .try_into()
+                    .map_err(|err| {
+                        Error::DbState(format!("couldn't convert report_id value: {err:?}"))
+                    })?;
+                let report_id = ReportId::from(report_id_bytes);
```

Comment:
As in `get_client_reports_for_task`, `RowExt::get_bytea_and_convert` could help here.

Reply:
I updated all instances of this pattern throughout `datastore.rs`.

Interestingly, the type parameter must be explicitly stated, even if it can be inferred. It took me a while to figure this out.